Update configuration docs (#429)
spiral/guide: docs/en/start/configuration.md

# Configuration

## Introduction

All of the configuration files for Spiral Framework are located in the `app/config` directory.

> These files contain options that allow you to configure your database connection, cache storage, sessions and more.

Reviewer: ```suggestion
These files contain options that allow you to configure for example, database connection, cache storage, sessions and more.
```

## Environment

It can be useful to have different configuration values depending on the environment in which the application is running.
For example, you may want a simple cache storage for local development in order to reduce the resource requirements of your testing environment, while a more robust and scalable storage may be preferred for production.

> To make it easy to manage different configuration values for different environments, Spiral uses the [DotEnv](https://github.com/vlucas/phpdotenv) PHP library.

Reviewer: Add link to the extension page `[DotEnv](../extension/dotenv.md)`

> To make it easy to manage different configuration values for different environments, Spiral uses the [DotEnv](https://github.com/vlucas/phpdotenv) PHP library.

Reviewer: ```suggestion
To make it easy to manage different configuration values for different environments, Spiral uses the [dotenv](../extension/dotenv.md) bridge.
```

In a new project installation, you will find a `.env.sample` file in the root directory of your application that defines several common environment variables.
When you install Spiral app, this file is automatically copied to `.env`.

Here is the example of `.env` in the application:

```dotenv
# Environment (prod or local)
APP_ENV=local

# Debug mode set to TRUE disables view caching and enables higher verbosity
DEBUG=true

# Verbosity level
VERBOSITY_LEVEL=verbose # basic, verbose, debug

# Set to an application specific value, used to encrypt/decrypt cookies etc

> ENCRYPTER_KEY=def000006531014288871c4c553d2b010e51e35a7f9550d6d23cecdd8b0729acdf9a0323f9d50a94c6d9166ef3ca6b931ac6b5579a71c4a32103b00ed64fa8987411238f

Reviewer: ```suggestion
ENCRYPTER_KEY={encrypt-key}
```

# Configuration

## Introduction

All of the configuration files for Spiral Framework are located in the `app/config` directory.
These files contain options that allow you to configure your database connection, cache storage, session lifetime, encryption key, and more.

> You can  add your configuration files.

Reviewer: ???

## Environment

It can be useful to have different configuration values depending on the environment in which the application is running.

> For example, you might want to use a different cache driver in your local development environment than you do on your production server.

Reviewer: than?

To make it easy to manage different configuration values for different environments, Spiral uses the [DotEnv](https://github.com/vlucas/phpdotenv) PHP library.
In a new project installation, you will find a `.env.sample` file in the root directory of your application that defines several common environment variables.

> When you install Spiral, this file is automatically copied to `.env`.

Reviewer: When you install Spiral app

}
```

> or via `env()` function

Reviewer: ```suggestion
or via a short function `env()`
```

```

> **Note**
> See the reference for each component configuration in the related documentation section.

## Accessing Configuration Values

### Config objects

In the Spiral Framework, all config objects are injectable.

> This means that when you try to resolve a config object through the container, it will automatically load all values from the config file or use default settings.

Reviewer: ```suggestion
This means that when you try to resolve a config object through the container, it will automatically load all values from the config file or will use default settings.
```
